assets/js/googlesitekit/datastore/site/notifications.test.js

```diff
@@ -21,18 +21,112 @@
  */
 import { actions, selectors } from './index';
 
+import {
+	createTestRegistry,
+	unsubscribeFromAll,
+} from '../../../../../tests/js/utils';
+import { CORE_SITE } from './constants';
+
 describe( 'core/site notifications', () => {
-	it( 'has appropriate notification actions', () => {
-		const actionsToExpect = [ 'addNotification', 'removeNotification' ];
-		expect( Object.keys( actions ) ).toEqual(
-			expect.arrayContaining( actionsToExpect )
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		const response = true;
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/site\/data\/mark-notification/,
+			{ body: JSON.stringify( response ), status: 200 }
 		);
 	} );
 
-	it( 'has appropriate notification selectors', () => {
-		const selectorsToExpect = [ 'getNotifications' ];
-		expect( Object.keys( selectors ) ).toEqual(
-			expect.arrayContaining( selectorsToExpect )
-		);
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'fetchMarkNotification', () => {
+			it( 'properly marks notifications', async () => {
```

Comment:
1. Let's split this test into two tests: one for `.not.toThrow` cases and one for `.toThrow` ones.
1. Also, let's use [`it.each( [ ... ] )( ... )`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) approach to avoid duplicating the logic multiple times.

Reply:
Please, use the same approach for other tests in this file.

Reply:
https://github.com/google/site-kit-wp/pull/4349/commits/5481de9105d5d34e365289ae10b3da3ee854b6f4